spec.html

```diff
@@ -11828,14 +11828,25 @@ <h1>Static Semantics: Early Errors</h1>
         <emu-grammar>PrimaryExpression : RegularExpressionLiteral</emu-grammar>
         <ul>
           <li>
-            It is a Syntax Error if BodyText of |RegularExpressionLiteral| cannot be recognized using the goal symbol |Pattern| of the ECMAScript RegExp grammar specified in <emu-xref href="#sec-patterns"></emu-xref>.
-          </li>
-          <li>
-            It is a Syntax Error if FlagText of |RegularExpressionLiteral| contains any code points other than `"g"`, `"i"`, `"m"`, `"s"`, `"u"`, or `"y"`, or if it contains the same code point more than once.
+            It is a Syntax Error if IsValidRegularExpressionLiteral(|RegularExpressionLiteral|) is *false*.
           </li>
         </ul>
       </emu-clause>
 
+      <emu-clause id="sec-primary-expression-regular-expression-literals-static-semantics-isvalidregularexpressionliteral" aoid="IsValidRegularExpressionLiteral">
+        <h1>Static Semantics: IsValidRegularExpressionLiteral ( literal )</h1>
+        <p>The abstract operation IsValidRegularExpressionLiteral determines if its argument is a valid regular expression literal. The following steps are taken:</p>
+        <emu-alg>
+          1. Assert: _literal_ is a Parse Node corresponding to a RegularExpressionLiteral.
+          1. If FlagText of _literal_ contains any code points other than `"g"`, `"i"`, `"m"`, `"s"`, `"u"`, or `"y"`, or if it contains the same code point more than once, return *false*.
+          1. Let _P_ be BodyText of _literal_.
+          1. If FlagText of _literal_ contains `"u"`,
+            1. If _P_ cannot be recognized using the goal symbol |Pattern[+U, +N]| of the ECMAScript RegExp grammar specified in <emu-xref href="#sec-patterns"></emu-xref>, return *false*.
+            1. Otherwise, return *true*.
+          1. Parse _P_ using the grammars in <emu-xref href="#sec-patterns"></emu-xref>. The goal symbol for the parse is |Pattern[~U, ~N]|. If the result of parsing contains a |GroupName|, reparse with the goal symbol |Pattern[~U, +N]| and use this result instead. If _P_ did not conform to the grammar, if any elements of _P_ were not matched by the parse, or if any Early Error conditions exist, return *false*. Otherwise, return *true*.
```

Comment:
This is a mouthful, and I can see it's basically duplicated over [here](https://tc39.github.io/ecma262/#sec-regexpinitialize). An improvement in the future might be to attempt some common factoring (but it might also be awkard due to runtime/static differences).

Reply:
I've found a way to factor out the common bits. I can submit that as a follow-up.

Reply:
@jmdyck If you still remember what your plan for factoring out the common bits is, this PR is now merged!

Reply:
I do, but it'll be easier (fewer collisions) if I wait until after PR #1552 is merged.

Reply:
@jmdyck #1552 landed; do want to submit that follow-up?

Reply:
Done: PR #1866.